pkg/ebpf/tracee.go

```diff
@@ -234,7 +214,21 @@ func New(cfg config.Config) (*Tracee, error) {
 	}
 	caps := capabilities.GetInstance()
 
-	// Pseudo events added by capture
+	// Initialize events state with mandatory events
+
+	t.eventsState[events.SchedProcessExec] = events.EventState{}
+	t.eventsState[events.SchedProcessExit] = events.EventState{}
+	t.eventsState[events.SchedProcessFork] = events.EventState{}
+
+	// Pseudo events added by control plane
+
+	t.eventsState[events.SignalCgroupMkdir] = policy.AlwaysSubmit
+	t.eventsState[events.SignalCgroupRmdir] = policy.AlwaysSubmit
+	t.eventsState[events.SignalSchedProcessFork] = policy.AlwaysSubmit
```

Comment:
Indeed. I have disabled only the userland portion, but having the Submit flag would cause the eBPF probes to be enabled. Nice catch!

Reply:
> We could even maybe couple dependent features like this.

I agree. We should have a way to tie "config option enabled" to "enabling the event". This is part of the refactor I started for the "EventState" but never finished (because of preemptions). Maybe I can try to tackle it together with the "event group + extensions" refactoring idea (which would be my next job after the proctree is done).